hydroengine_service/dgds_functions.py

```diff
@@ -0,0 +1,427 @@
+import ee
+import json
+import logging
+import numpy as np
+import os
+
+from hydroengine_service import config
+from hydroengine_service import error_handler
+
+EE_CREDENTIALS = ee.ServiceAccountCredentials(config.EE_ACCOUNT,
+                                              config.EE_PRIVATE_KEY_FILE)
+
+ee.Initialize(EE_CREDENTIALS)
+
+# visualization parameters for datasets
+APP_DIR = os.path.dirname(os.path.realpath(__file__))
+DATASET_DIR = os.path.join(APP_DIR, 'datasets')
+with open(DATASET_DIR + '/dataset_visualization_parameters.json') as json_file:
+    DATASETS_VIS = json.load(json_file)
+
+logger = logging.getLogger(__name__)
+
+def get_dgds_source_vis_params(source, image_id=None):
+    """
+    Check source and/or image_id has default visualization parameters defined
+    :param source: String, source/location of Earth Engine Object
+    :param image_id: String, Earth Engine Image id
+    :return:
+    """
+    data_params = DATASETS_VIS.get(source, None)
+    if image_id and not data_params:
+        data_params = DATASETS_VIS.get(image_id, None)
+    assert data_params, f'{source} not in assets.'
+    return data_params
+
+
+def get_dgds_data(source,
+                  dataset=None,
+                  image_id=None,
+                  band=None,
+                  function=None,
+                  start_date=None,
+                  end_date=None,
+                  image_num_limit=None):
+    """
+
+    :param source: String, source/location of Earth Engine Object
+    :param dataset: String, name of Earth Engine Object dataset
+    :param image_id: String, Earth Engine Image id
+    :param band: String, band name to select from image
+    :param function: String, function to apply to image
+    :param start_date: String, start date to filter collection on
+    :param end_date: String, end date to filter collection on
+    :param image_num_limit: String, limit of objects to return
+    :return: Dictionary
+    """
+    # Get list of objects with imageId and date for collection
+    data_params = get_dgds_source_vis_params(source, image_id)
+    info = get_image_collection_info(source, start_date, end_date, image_num_limit)
+    if not info:
+        return
```

Comment:
This is... odd? It seems that you handle invalid input by going down into functions where they sometimes just return None and handle that on the top level.

You can have some logging here, but it's nicer to just raise an error asap and possible catch it at the highest level if you need a sensible 500 error message for the client.